apps/angular/5-crud-application/src/services/crud-special.service.ts

```diff
@@ -0,0 +1,63 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
+import { Observable, throwError } from 'rxjs';
+import { catchError, retry } from 'rxjs/operators';
+import { TodoDetails } from './../model/user';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CrudSpecialService {
+  constructor(private http: HttpClient) {}
+
+  getAll$ = this.http
+    .get<TodoDetails[]>('https://jsonplaceholder.typicode.com/todos')
+    .pipe(
+      retry(3),
+      catchError((error) => {
+        console.error(error);
+        return throwError(() => error);
+      }),
+    );
+
+  public allTodos = toSignal(this.getAll$, {
```

Comment:
you should never use `toSignal` inside a service. 
ToSignal is linked to the life of a component. 

Each time you will call `allTodos`, a new subscription will be created. 

`ToSignal` is the last step of your reactive chain, in the component.